easybuild/framework/extension.py

```diff
@@ -68,6 +68,22 @@ def __init__(self, mself, ext):
         self.patches = self.ext.get('patches', [])
         self.options = copy.deepcopy(self.ext.get('options', {}))
 
+        if extra_params:
+            self.cfg.extend_params(extra_params, overwrite=False)
```

Comment:
To be honest, I'm not sure. It was added like this in #1498, and even though I can't think of a reason for it now, there must have been a good reason since support for not overwriting was added to `extend_params` in the same go. That reason may no longer be valid now though...

I'll leave this as is for now, but this is probably a bug since this means that the `runtest` easyconfig parameter that is hard set to `True` in `PythonPackage` (while the default is `False`) is not being set when Python packages are installed as extensions.
It's not a huge problem, but it does mean that the `scipy` tests are not being run when `scipy` is installed as an extension...

The only case where overwriting matters or not is when the easyblock that is used for installing the extension redefines a general easyconfig parameter, which only occurs for `runtest` for stuff like Python packages, at first sight, and probably only affects `scipy` (the `numpy` tests are always run, regardless of `runtest`, which is also a bug I guess).

Follow-up in #2195.